File: desktop/src/types/stats.ts
export interface LlmUsageStats {
  total_calls: number
  success_calls: number
  failed_calls: number
  total_tokens: number
  total_cost: string
  avg_latency_ms: number | null
}

export interface PurposeUsageStats {
  purpose: string
  calls: number
  success_calls: number
  total_tokens: number
  total_cost: string
}

export interface ModelUsageStats {
  model: string
  calls: number
  total_tokens: number
  total_cost: string
}

export interface TaskCostStats {
  task_id: string
  task_name: string
  calls: number
  total_tokens: number
  total_cost: string
}

export interface DailyUsageStats {
  date: string
  calls: number
  total_tokens: number
  total_cost: string
}

export interface TaskFlowStats {
  total_runs: number
  completed_runs: number
  failed_runs: number
  success_rate: number | null
  avg_duration_seconds: number | null
}

export interface StatsSummary {
  llm: LlmUsageStats
  tasks: TaskFlowStats
  by_purpose: PurposeUsageStats[]
  by_model: ModelUsageStats[]
  by_task: TaskCostStats[]
  daily: DailyUsageStats[]
}

File: backend/app/schemas/stats.py
import uuid
from datetime import date
from decimal import Decimal

from pydantic import BaseModel


class LlmUsageStats(BaseModel):
    total_calls: int = 0
    success_calls: int = 0
    failed_calls: int = 0
    total_tokens: int = 0
    total_cost: Decimal = Decimal("0")
    avg_latency_ms: float | None = None


class PurposeUsageStats(BaseModel):
    purpose: str
    calls: int = 0
    success_calls: int = 0
    total_tokens: int = 0
    total_cost: Decimal = Decimal("0")


class ModelUsageStats(BaseModel):
    model: str
    calls: int = 0
    total_tokens: int = 0
    total_cost: Decimal = Decimal("0")


class TaskCostStats(BaseModel):
    task_id: uuid.UUID
    task_name: str
    calls: int = 0
    total_tokens: int = 0
    total_cost: Decimal = Decimal("0")


class DailyUsageStats(BaseModel):
    date: date
    calls: int = 0
    total_tokens: int = 0
    total_cost: Decimal = Decimal("0")


class TaskFlowStats(BaseModel):
    total_runs: int = 0
    completed_runs: int = 0
    failed_runs: int = 0
    success_rate: float | None = None
    avg_duration_seconds: float | None = None


class StatsSummary(BaseModel):
    llm: LlmUsageStats
    tasks: TaskFlowStats
    by_purpose: list[PurposeUsageStats]
    by_model: list[ModelUsageStats]
    by_task: list[TaskCostStats]
    daily: list[DailyUsageStats]
